### PR TITLE
Relax constraints on sqlalchemy

### LINKED ISSUE #4: Constraint on sqlalchemy seems to prevent installing Parsl with python>=3.8
The constraint to `sqlalchemy ==1.3.3`:
https://github.com/conda-forge/parsl-feedstock/blob/master/recipe/meta.yaml#L36
seems to prevent installation of `parsl` with `python >=3.8`.  
If I do:
```
$ conda search --info sqlalchemy=1.3.3
```
I see support only for python 2.7, 3.6 and 3.7.

It seems like this constraint isn't needed, based on the upstream requirements (`sqlalchemy>=1.3.0,!=1.3.4`):
https://github.com/Parsl/parsl/blob/1.0.0/setup.py#L11

If you are open to it, I would like to make a PR for a new build with the same constraints on `sqlalchemy` as upstream.  I wanted to make sure this was okay before making the PR.

### PR DESCRIPTION
**Pull request by xylar**
Also:
* add @xylar as a maintainer
* add some other dependency constraints to match upstream
* fix capitalization of `flask`

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

closes #4

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,20 +24,20 @@ requirements:
...
     - globus-sdk
     - dill
     - tblib
     - ipykernel
     - requests
     - paramiko
-    - psutil
-    - sqlalchemy ==1.3.3
+    - psutil >=5.5.1
+    - sqlalchemy >=1.3.0,!=1.3.4
```
Comment (xylar) [inline recipe/meta.yaml:36]: @conda-forge/parsl, this is the main change, so make sure you agree that this is correct.  The other package constraints should be the same as upstream as well, but feel free to check.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -63,3 +63,4 @@ extra:
   recipe-maintainers:
     - ZhuozhaoLi
     - yadudoc
+    - xylar
```
Comment (xylar) [inline recipe/meta.yaml:66]: I'm happy to *not* add myself as a maintainer if you prefer to take care of this package yourselves.  Just volunteering to help if you want it.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -24,20 +24,20 @@ requirements:
...
     - paramiko
-    - psutil
-    - sqlalchemy ==1.3.3
+    - psutil >=5.5.1
+    - sqlalchemy >=1.3.0,!=1.3.4
     - sqlalchemy-utils
     - pydot
     - networkx
-    - Flask
+    - flask >=1.0.2
```
Comment (yadudoc) [inline recipe/meta.yaml:40]: @ZhuozhaoLi Can you check if this is right? I know we've had issues with `flask` before.

File: recipe/meta.yaml (hunk 1 above)
Comment (yadudoc) [reply recipe/meta.yaml:36]: This looks good.

File: recipe/meta.yaml (hunk 2 above)
Comment (yadudoc) [reply recipe/meta.yaml:66]: I'm happy you added yourself, we wish more of the contributors do this more often.

**Review by yadudoc — COMMENTED** [review]
I'm waiting on @ZhuozhaoLi to confirm that the `flask` requirements line doesn't break things. Otherwise, everything here looks good to go.

File: recipe/meta.yaml (hunk 3 above)
Comment (ZhuozhaoLi) [reply recipe/meta.yaml:40]: Yes, this is right.

**Comment by xylar** [comment]
Okay, we'll merge whenever you're both happy.

**Review by ZhuozhaoLi — APPROVED** [review]
This looks good. Thanks!

**Pull request merged**